test/prepop/MemberPrePopulationProcessorSpec.scala

```diff
@@ -0,0 +1,822 @@
+import play.api.libs.json._
+
+import scala.util.Success
+
+class MemberPrePopulationProcessorSpec extends BaseSpec with TestValues {
+
+  private val processor = new MemberPrePopulationProcessor()
+
+  "MemberPrePopulationProcessor" - {
+
+    "clean" - {
+      "should cleanup the Member details from baseReturn and put it onto currentUA" in {
+        val currentUa = emptyUserAnswers.unsafeSet(__ \ "current", JsString("dummy-current-data"))
+        val result = processor.clean(
+          baseUA = emptyUserAnswers.copy(data = SensitiveJsObject(baseReturn.as[JsObject])),
+          currentUA = currentUa
+        )
+        result mustBe Success(
+          currentUa.copy(data = SensitiveJsObject(cleanResultJsValue.as[JsObject]))
+        )
+      }
+    }
+  }
+}
+
+object MemberPrePopulationProcessorSpec {
+
+  val baseReturn: JsValue =
+    Json.parse("""
+        |{
+        |  "membersPayments": {
+        |    "recordVersion": "004",
+        |    "unallocatedContribsMade": true,
+        |    "unallocatedContribAmount": {
+        |      "value": 123.45,
+        |      "displayAs": "123.45"
+        |    },
+        |    "memberDetails": {
+        |      "personalDetails": {
+        |        "nameDob": {
+        |          "0": {
+        |            "firstName": "Ferdinand",
+        |            "lastName": "Bull",
+        |            "dob": "1960-05-31"
+        |          },
+        |          "1": {
+        |            "firstName": "Johnny",
+        |            "lastName": "Quicke",
+        |            "dob": "1940-10-31"
+        |          },
+        |          "2": {
+        |            "firstName": "Foury",
+        |            "lastName": "Lasty",
+        |            "dob": "1940-10-31"
+        |          },
+        |          "3": {
+        |            "firstName": "Five",
+        |            "lastName": "Last",
+        |            "dob": "1940-10-31"
+        |          },
+        |          "4": {
+        |            "firstName": "Six",
+        |            "lastName": "User",
+        |            "dob": "1949-12-31"
+        |          },
+        |          "5": {
+        |            "firstName": "NewUser",
+        |            "lastName": "LastName",
+        |            "dob": "1940-10-31"
+        |          }
+        |        },
+        |        "nationalInsuranceNumber": {
+        |          "0": true,
+        |          "1": false,
+        |          "2": false,
+        |          "3": false,
+        |          "4": false,
+        |          "5": false
+        |        },
+        |        "nino": {
+        |          "0": "EB103145A"
+        |        },
+        |        "memberDetailsSectionCompleted": {
+        |          "0": {},
+        |          "1": {},
+        |          "2": {},
+        |          "3": {},
+        |          "4": {},
+        |          "5": {}
+        |        },
+        |        "memberStatus": {
+        |          "0": "Changed",
```

Comment:
if the status is deleted, member should not be copied over.

Reply:
the records with `memberStatus : Deleted` was already marked as `"soft-deleted"` in baseReturn json in retrieval service and we aren't copying over these soft-deleted member records.

Reply:
Do you need to test that soft-deleted members are not copied over?

Reply:
I put `"soft-deleted"` into baseReturn and verified that it's not within copied over userAnswer after cleanup.